test/contrib/test_funsor.py

```diff
@@ -196,6 +197,14 @@ def testing():
         testing()
 
 
+def test_nested_plate():
+    with enum(first_available_dim=-3):
+        with enum_plate("a", 5):
+            with enum_plate("b", 2):
+                x = numpyro.sample("x", dist.Normal(0, 1), rng_key=random.PRNGKey(0))
+                assert x.shape == (2, 5)
```

Comment:
Was this assertion false before this PR?

Reply:
Yes, actually, there is an error that happens even before this assertation. Currently, plate "a" will have `dim=-1`, plate "b" will also have `dim=-1`, this makes it impossible to expand `Normal(0, 1)` to both (5,) and (2,) shapes.

Reply:
I guess I'm puzzled at how the change in this PR fixes that - as far as I can tell, setting aside `subsample`/`param` statements, the only behavioral difference is that the logic in `plate.process_message` now runs for `"plate"` sites and `super().process_message(msg)` now does not run for `"plate"` sites, since the bulk of the code in `plate.process_message` was copied verbatim from `numpyro.primitives.plate`.

Seems like I really should update `numpyro.contrib.funsor` ASAP.